numpy/lib/function_base.py

```diff
@@ -3612,17 +3786,81 @@ def _quantile_is_valid(q):
     return True
 
 
-def _quantile_ureduce_func(a, q, axis=None, out=None, overwrite_input=False,
-                           interpolation='linear', keepdims=False):
+def _validate_weights(a, q, axis, weights):
+    if weights is None:
+        wgt = None
```

Comment:
`return None`, would let you unindent everything below

Reply:
Good point